src/main/java/com/unzer/payment/paymenttypes/Applepay.java

```diff
@@ -33,116 +33,135 @@
-		this.version = version;
-	}
-
-	public String getData() {
-		return data;
-	}
-
-	public void setData(String data) {
-		this.data = data;
-	}
-	
-	public String getSignature() {
-		return signature;
-	}
-
-	public void setSignature(String signature) {
-		this.signature = signature;
-	}
-
-	public ApplepayHeader getHeader() {
-		return header;
-	}
-
-	public void setHeader(ApplepayHeader header) {
-		this.header = header;
-	}
-
-	public String getCurrencyCode() {
-		return currencyCode;
-	}
-
-	public void setCurrencyCode(String currencyCode) {
-		this.currencyCode = currencyCode;
-	}
-
-	public BigDecimal getTransactionAmount() {
-		return transactionAmount;
-	}
-
-	public void setTransactionAmount(BigDecimal transactionAmount) {
-		this.transactionAmount = transactionAmount;
-	}
-
-	public String getNumber() {
-		return number;
-	}
-
-	public void setNumber(String number) {
-		this.number = number;
-	}
-
-	public String getExpiryDate() {
-		return expiryDate;
-	}
-
-	public void setExpiryDate(String expiryDate) {
-		this.expiryDate = expiryDate;
-	}
-
+    private String version;
+    private String data;
+    private String signature;
+    private ApplepayHeader header;
+
+    private String method;
+    private String applicationPrimaryAccountNumber;
+    private String applicationExpirationDate;
+    private String currencyCode;
+    private BigDecimal transactionAmount;
+
+    public Applepay() {
+    }
+
+    public Applepay(String version, String data, String signature, String currencyCode) {
+        this.version = version;
+        this.data = data;
+        this.signature = signature;
+        this.currencyCode = currencyCode;
+    }
+
+    @Override
+    public String getTypeUrl() {
+        return "types/applepay";
+    }
+
+    @Override
+    public PaymentType map(PaymentType applepay, JsonObject jsonApplePay) {
+        ((Applepay) applepay).setId(jsonApplePay.getId());
+        ((Applepay) applepay).setApplicationExpirationDate(((JsonApplepayResponse) jsonApplePay).getApplicationExpirationDate());
+        ((Applepay) applepay).setApplicationPrimaryAccountNumber(((JsonApplepayResponse) jsonApplePay).getApplicationPrimaryAccountNumber());
+        ((Applepay) applepay).setCurrencyCode(((JsonApplepayResponse) jsonApplePay).getCurrencyCode());
+        ((Applepay) applepay).setTransactionAmount(((JsonApplepayResponse) jsonApplePay).getTransactionAmount());
+        ((Applepay) applepay).setRecurring(((JsonApplepayResponse) jsonApplePay).getRecurring());
+        GeoLocation tempGeoLocation = new GeoLocation(((JsonApplepayResponse) jsonApplePay).getGeoLocation().getClientIp(), ((JsonApplepayResponse) jsonApplePay).getGeoLocation().getCountryIsoA2());
+        ((Applepay) applepay).setGeoLocation(tempGeoLocation);
+        return applepay;
+    }
+
+    public Authorization authorize(BigDecimal amount, Currency currency, URL returnUrl) throws HttpCommunicationException {
+        return authorize(amount, currency, returnUrl, null);
```

Comment:
"getUnzer()." expected here?

Reply:
No. This function calls the class-internal `authorize`-Function and is passing `null` as the `Customer`-Parameter to it. So there wouldn't be a huge difference in using `getUnzer()` directly.